integration/legacy_migrate.go

```diff
@@ -186,7 +186,7 @@ func translateGossipPrefix(p byte) byte {
 	return p
 }
 
-func migrateLegacyDBs(chaindataDir string, dbs kvdb.FlushableDBProducer) error {
+func migrateLegacyDBs(chaindataDir string, dbs kvdb.FlushableDBProducer, mode string) error {
 	if !isEmpty(path.Join(chaindataDir, "gossip")) {
```

Comment:
I think this check is not much necessary due to we already check it before calling the function `migrateLegacyDBs()`; it's okay to re-check inside the function but I prefer to use short circuit as below to reduce one long nested if condition

```suggestion
	if isEmpty(path.Join(chaindataDir, "gossip")) {
	     return null
	}
```